### ISSUE TITLE
Any plans for fully implementing Future Payments in PayPal-Python-SDK?

### ISSUE THREAD

**Issue by helgihg** [paypal/PayPal-Python-SDK#200]
I'm sorry if this is the improper place to ask. Please direct me to the proper channels if I'm asking in the wrong place, but my question regards PayPal-Python-SDK specifically, not PayPal as a whole, so I figure this is the best place to start.

We're interested in using Future Payments, but the documentation seems to assume mobile development for that feature. We're not, however, developing a mobile app and have no reason to create one.

Specifically, in order to do Future Payments, an authorization code using OAuth2 is required. There is nothing mobile-specific about OAuth2 so I wonder why it's mobile-specific at PayPal. Information on the matter seems to be scarce, and it is unclear whether this is a temporary arrangement or a design choice.

So what I'm wondering, ultimately is this: Is full support for Future Payments planned for PayPal-Python-SDK in the near future?

https://developer.paypal.com/docs/integration/mobile/make-future-payment/#get-an-auth-code

**Comment by xiaoleih41** [comment]
As far as I know, there will  be no plan to add it to the SDK. We have other products available may fit your business needs. Can you tell me what exact you want to accomplish? Will [billing plan/billing agreement](https://developer.paypal.com/docs/integration/direct/billing-plans-and-agreements/) or [PayPal credit product](https://www.paypal.com/us/webapps/mpp/paypal-credit) work for your needs?

**Comment by helgihg** [comment]
Thanks for the response! I would really like to know why it's only available in the mobile SDK, but no one seems to know. I think it would save a lot of people's time just elaborating on why it is that way, because a lot of people are asking about this online but either no one knows or everyone who does is reluctant to explain it (which in itself would be interesting).

But anyway, to the problem at hand. We offer subscriptions that vary in price and duration. The price of the first month depends on how much is left when the order is placed, and the user may either order additional services or change options of his existing subscriptions at any given time in the future. If the user does so, then the same rule applies, that the first month's price is determined by how much is left of the month.

Furthermore, we would like the user to be able to choose a previously used payment method when ordering additional services, as opposed to creating an entirely new payment method every time. Only Future Payments seem to make this possible, whereas Billing Plans and Billing Agreements appear to require the user's authentication every time it is changed.

In short, Billing Plans and Billing Agreements seem too rigid for this.

If I'm mistaken about any of this, I'd be be quite happy to be corrected.

**Comment by xiaoleih41** [comment]
I agree with you that the current billing plan/agreement doesn't work in situations like this. However there will be some new features in development that will possibly provide the type of flexibility that you are looking for.  I am not given any exact timeline but I know it's something planned  and prepared.

**Comment by bluk** [comment]
@helgihg You may want to look into [Braintree's Recurring Billing](https://developers.braintreepayments.com/guides/recurring-billing/overview) for your use case. Braintree (a PayPal service) accepts PayPal and other payment methods. You can email their support on the bottom of that page to ensure your use case is supported.

**Comment by helgihg** [comment]
@bluk Thanks for the tip! @xiaoleih41 Thanks for the info! I'll keep hoping. :)

**Comment by xiaoleih41** [comment]
I am closing thread as the question has been answered. Please keep an eye on the PayPal product release by the end of this year if [Braintree's Recurring Billing](https://developers.braintreepayments.com/guides/recurring-billing/overview) doesn't work for you either.  REST API Billing Agreement has a field called [`setup_fee`](https://developer.paypal.com/docs/api/payments.billing-agreements/?mark=setup_fee#definition-merchant_preferences) It allows you to instantly charge an amount when the buyer approves the billing agreement. I believe this should give you the flexibility to charge the first partial monthly fee. However it won't let you change the monthly fee down the road.
